{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/utils/sites.py

```diff
@@ -1,8 +1,16 @@
+from urllib.parse import urlparse
+
 from django.conf import settings
 
 
 def get_site_url():
-    """Get the full base URL for this site based on settings."""
-    protocol = "http" if settings.IN_DEV else "https"
-    domain = settings.CURRENT_DOMAIN
-    return "{0}://{1}{2}".format(protocol, domain, f":{settings.CURRENT_PORT}" if settings.CURRENT_PORT else "")
+    if not (settings.HEROKU_APP_NAME or settings.CURRENT_DOMAIN):
+        raise Exception("Environment variable for either HEROKU_APP_NAME or CURRENT_DOMAIN must exist")
+    domain = f"{settings.HEROKU_APP_NAME}.herokuapp.com" if settings.HEROKU_APP_NAME else settings.CURRENT_DOMAIN
+    domain = domain.strip("/").split("/")[-1]
+    scheme = "http://" if settings.IN_DEV else "https://"
+    site_url = urlparse(f"{scheme}{domain}")
+    if settings.CURRENT_PORT:
+        netloc = site_url.netloc + f":{settings.CURRENT_PORT}"
+        site_url = site_url._replace(netloc=netloc)
+    return site_url.geturl()
```

Comment:
The will now generate the correct URL for Heroku Review Apps
In addition, several edge case bugs are now handled. Don't have to worry about CURRENT_DOMAIN needing https are the beginning or not.